src/app/administrator/admin-publisher/admin-publisher.component.ts

```diff
@@ -1,15 +1,63 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit } from "@angular/core";
+import { OLmsService } from "src/app/common/o/services/oLms.service";
+import { environment } from "src/environments/environment";
 
 @Component({
-  selector: 'app-admin-publisher',
-  templateUrl: './admin-publisher.component.html',
-  styleUrls: ['./admin-publisher.component.css']
+  selector: "app-admin-publisher",
+  templateUrl: "./admin-publisher.component.html",
+  styleUrls: ["./admin-publisher.component.css"],
 })
 export class AdminPublisherComponent implements OnInit {
+  publishers: any[];
+  books: any[];
 
-  constructor() { }
+  constructor(private lmsService: OLmsService) {}
 
   ngOnInit() {
+    this.readPublishers();
+    this.readBooks();
   }
 
+  readPublishers() {
+    this.lmsService
+      .get(`${environment.adminBackendUrl}${environment.readPublisherUri}`)
+      .subscribe(
+        (result: object[]) => (this.publishers = result),
+        (error: any) => {
+          // do something with a logger here
+          this.publishers = [];
+          alert(error);
+        }
+      );
+  }
+
+  readBooks() {
+    this.lmsService
+      .get(`${environment.adminBackendUrl}${environment.readBookUri}`)
+      .subscribe(
+        (result: object[]) => (this.books = result),
+        (error: any) => {
+          // do something with a logger here
+          this.books = [];
+          alert(error);
+        }
+      );
+  }
+
+  deletePublisher(publisherId: number) {
+    if (!confirm("Are you sure you want to delete this publisher?")) return;
```

Comment:
This is a clean way to accomplish the "are you sure?".  I am guessing this looks just like the alert, but with a yes/no button as well?